src/script/conversation/ConversationRepository.ts

```diff
@@ -1306,7 +1306,9 @@ export class ConversationRepository {
     const connection = user.connection();
     if (connection) {
       this.logger.log(`There's a connection with user ${userId.id}, getting a 1:1 conversation for the connection`);
-      return this.get1to1ConversationForConnection(connection, options);
+      const conversation = await this.get1to1ConversationForConnection(connection, options);
+      // In case we got a conversation back, we make sure the participating user entities are up to date
+      return conversation ? this.updateParticipatingUserEntities(conversation) : null;
```

Comment:
Could this be checked inside the method returning the conversation entity already?

Reply:
Well I wanted the side effect to be as high as possible in the call stack. As much as possible I try to isolate side effects to the highest level possible. 

This seemed like the right layer to do this. 
Also makes the `get1to1ConversationForConnection` more focused (it just gets the conversation and doesn't try to be too clever about it)

Reply:
Makes sense 👌